features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -98,6 +99,17 @@ internal class DatadogRumMonitor(
 
     // region RumMonitor
 
+    override val currentSessionId: String?
+        get() {
+            val applicationScope = rootScope as? RumApplicationScope
+            val sessionScope = applicationScope?.activeSession as? RumSessionScope
+            if (sessionScope != null && sessionScope.sessionState == RumSessionScope.State.NOT_TRACKED) {
+                // Aligns with iOS which assigns a NULL session id when the session is not tracked
+                return RumContext.NULL_UUID
```

Comment:
For "readability" purposes, in link with Nikita's comment above, maybe we should return `null` instead here. This would make it easier for customers to not use a `"000…"` value which doesn't make sense?